README.md

```diff
@@ -399,6 +399,44 @@ var source = KafkaConsumer.PlainPartitionedManualOffsetSource(consumerSettings,
 
 Are not implemented yet. Waiting for issue https://github.com/akkadotnet/Akka.Streams.Kafka/issues/85 to be resolved.
 
+### Partition Events handling
+
+Sometimes you may need to add custom handling for partition events, like assigning partition to consumer. To do that, you will need:
+
+1. To write custom implementation of `IPartitionEventHandler` interface:
+```c#
+class CustomEventsHandler : IPartitionEventHandler
+{
+	/// <inheritdoc />
+	public void OnRevoke(IImmutableSet<TopicPartitionOffset> revokedTopicPartitions, IRestrictedConsumer consumer)
+	{
+		// Your code here
```

Comment:
How can we use this to ensure that:

1. the stream has been drained
2. it is safe to clear all state in the actor system (for example, kill all of the shadow actors) before allowing reassignment?

Can you add details around the timing of when `OnRevoke` is called in relation to the stream? When this is called, does it mean that no more new messages are being sent by the source?

One potential solution I could take would be to clean up state within `OnAssign` instead. Thoughts?

Reply:
Well... Let's assume that you are using only one partition that is assigned/revoked from your consumer (otherwise you will need to keep track of the partitions assigned/revoked). 

I do not fully understand your use case, but OnRevoke is called once kafka tells consumer that partition is revoked - which means that you are not going to receive more messages from kafka anymore. 
But - since all of the stuff here is eventually consistent, in theory you can receive some messages from the stream after `OnRevoke` was invoked. Our internal kafka consumer will not receive messages anymore, but some messages could already be received / buffered and on the half way to your actors.

So you can not rely on `OnRevoke`  as a signal that stream is drained. But you have `revokedTopicPartitions` which contain partitions offsets. I think you can wait until you receive message that has an offset matching revoked partition offset - that is going to be the last message from partition. And here you will not receive new messages until next partition assignment.

Doing cleanup on `OnAssign` method also sounds good to me. But if cleanup make take a while, you should probably do that in background, and make sure that you do not start handling new messages until cleanup is finished.